packages/ckeditor5-clipboard/src/clipboardmarkersutils.ts

```diff
@@ -536,16 +589,25 @@ export default class ClipboardMarkersUtils extends Plugin {
 export type ClipboardMarkerRestrictedAction = 'copy' | 'cut' | 'dragstart';
 
 /**
- * Specifies copy, paste or move marker restrictions in clipboard. Depending on specified mode
- * it will disallow copy, cut or paste of marker in clipboard.
- *
- * 	* `'default'` - the markers will be preserved on cut-paste and drag and drop actions only.
- * 	* `'always'` - the markers will be preserved on all clipboard actions (cut, copy, drag and drop).
- * 	* `'never'` - the markers will be ignored by clipboard.
+ * Specifies behavior of markers during clipboard actions.
  *
  * @internal
  */
-export type ClipboardMarkerRestrictionsPreset = 'default' | 'always' | 'never';
+export type ClipboardMarkerConfiguration = {
+	allowedActions: NonEmptyArray<ClipboardMarkerRestrictedAction> | 'all';
+
+	// If false, do not copy marker when only part of its content is selected.
+	withPartiallySelected?: boolean;
+
+	// If true then every marker that is present in clipboard document fragment element obtain new generated id just before pasting.
+	// It means that it is possible to perform copy once and then paste it multiple times wherever we want.
+	//
+	// On the other hand if it has false value the marker will be not pasted because ID already exists in the document.
+	//
+	// This flag is ignored in `cut` and `dragstart` actions because source marker is moved to graveyard and
```

Comment:
Hmm.. Is it true? We use this flag on `paste`.

Reply:
Yes, in `cut` / `dragstart` actions source element is moved to `graveyard` and `isInGraveyard` is always true:

```if ( copyMarkerConfig.regenerateMarkerIdsOnPaste || isInGraveyard ) {```